webhook/resourcesemantics/defaulting/defaulting_test.go

```diff
@@ -543,5 +714,27 @@ func newTestResourceAdmissionController(t *testing.T) webhook.AdmissionControlle
 		ctx, testResourceValidationName, testResourceValidationPath,
 		handlers, func(ctx context.Context) context.Context {
 			return ctx
-		}, true).Reconciler.(*reconciler)
+		}, true, callbacks).Reconciler.(*reconciler)
+}
+
+func resourceCallback(_ context.Context, uns *unstructured.Unstructured) error {
+	var resource Resource
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uns.UnstructuredContent(), &resource); err != nil {
```

Comment:
Can we have a more representative callback?

Consumers shouldn't be decoding unstructured objects to typed structs in these callbacks. A solution already exists for that.

These callbacks are really about using unstructured types to set values to a few fields without being tied to a specific type or version.

Reply:
> Can we have a more representative callback?
>
> Consumers shouldn't be decoding unstructured objects to typed structs in these callbacks. A solution already exists for that.
>
> These callbacks are really about using unstructured types to set values to a few fields without being tied to a specific type or version.

I agree, this is just a test callback.

I don't understand the goal of having a callback dealing with unstructured things in a test that we know is using the `Resource` object.

Reply:
I guess we want to assert the arguments being passed to the callback etc..